### DISCUSSION TITLE
using chakra-ui in micro-frontends

### DISCUSSION THREAD

**Discussion by csigmanek** [chakra-ui/chakra-ui#5025] [General] [answered]
Hello, we were planning on using chakra-ui in micro-frontends. However, chakra is setting agressively global styles such as

`.h5, h5 { margin: 0}`

Is there a way to wrap these styles into a class? Like this:

`.myapp { h5, .h5 {margin: 0}}`

What is the recommended way to disable or better wrap the global styles? It's breaking other UI-Components. (Which are maintained by other companies that are not using chakra for some reasons.)

**Comment by dmitriifrlv** [comment, marked as answer]
You can wrap only the part of your app with `<ChakraProvider resetCSS={false}>`. Setting `resetCSS={false}` will help to preserve global styles from resetting and at the same time you should get nice looking chakra-ui components inside `ChakraProvider`. 

[Demo](https://codesandbox.io/s/using-chakra-ui-components-without-state-reset-q4qm2?file=/src/App.js:140-173)

**Reply by csigmanek** [reply]
Thank you. Worked as you've described it.

**Comment by primos63** [comment]
You are looking for something where the Chakra styles are constrainted to a particular piece of software/portion of your page (scoped). Is that correct?

**Reply by csigmanek** [reply]
I was simply hoping not to overwrite styles of my container app anymore. The trick dmitriifrlv mentioned helped.
